core/string/char_utils.h

```diff
@@ -33,6 +33,1444 @@
 
 #include "core/typedefs.h"
 
+struct CharRange {
+	char32_t start;
+	char32_t end;
+};
+
+static CharRange xid_start[] = {
+	{ 0x41, 0x5a },
+	{ 0x5f, 0x5f },
+	{ 0x61, 0x7a },
+	{ 0xaa, 0xaa },
+	{ 0xb5, 0xb5 },
+	{ 0xba, 0xba },
+	{ 0xc0, 0xd6 },
+	{ 0xd8, 0xf6 },
```

Comment:
Maybe these could go to a `core/string/char_range.inc` to reduce the clutter in `char_utils.h` (which would be included here for the two new methods that use this data)? Having so much data in a header makes the actual logic harder to consult IMO.

Reply:
Moved ranges to a separate file.